openmdao/recorders/sqlite_recorder.py

```diff
@@ -64,13 +79,17 @@ def record_metadata(self, group):
             `System` containing vectors
         """
 
-        params = group.params.iteritems()
-        #resids = group.resids.iteritems()
-        unknowns = group.unknowns.iteritems()
-
-        self.out_metadata['format_version'] = format_version
-        self.out_metadata['Parameters'] = dict(params)
-        self.out_metadata['Unknowns'] = dict(unknowns)
+        if MPI and MPI.COMM_WORLD.rank > 0 :
+            raise RuntimeError("not rank 0")
+        else:
+            params = group.params.iteritems()
+            #resids = group.resids.iteritems()
+            unknowns = group.unknowns.iteritems()
+            self.out_metadata['format_version'] = format_version
+            self.out_metadata['Parameters'] = dict(params)
+            self.out_metadata['Unknowns'] = dict(unknowns)
+            self.out_metadata['system_metadata'] = group.metadata
+            self.out_metadata['model_viewer_data'] = self.required_data
```

Comment:
suggest that code would be more clear and consistent if `model_viewer_data` and `required_data` were just called `model_data` everywhere they occur